src/screens/common/RegisterEvent/RegisterEvent.tsx

```diff
@@ -27,22 +28,25 @@ const RegisterEvent: React.FC<RegisterEvent> = ({ eventId, userId }) => {
   } = useForm();
   const [
     createInvitation,
-    { data, error, loading },
+    { error, loading },
   ] = useMutation<CreateInvitation>(CREATE_INVITATION);
-
-  // TODO Make register event/create invitation work
+  const toastProvider = useToastProvider();
+  
   const onSubmit = useCallback(async (data) => {
     try {
-      await createInvitation({
+      const { data: invitationData } = await createInvitation({
         variables: {
           createInvitationEventId: eventId,
           createInvitationUserId: userId,
           createInvitationNumAttendees: parseInt(data.dependents),
         },
       });
+      if (invitationData) {
+        setCurrentInvitationId(invitationData.id);
+      }
       navigation.navigate(SCREEN_NAMES.common.events.eventDetails);
     } catch (e) {
-      // Do nothing
+      toastProvider.showError("Failed to register to event");
```

Comment:
Based on the docs, I don't think the `e` that's being thrown by this is the error thrown by the BE. They usually handle the errors on the error object returned by the useMutation hook. That's why we handle our errors in the useEffect hook.

(Same applies to the implementation in EventDetails)

Reply:
Also, would be nice to attach the error message coming from BE on the failure message so that we can debug it easily.